Source/DafnyDriver/CSVTestLogger.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
+
+namespace Microsoft.Dafny {
+  public class CSVTestLogger : ITestLoggerWithParameters {
+
+    private readonly List<TestResult> results = new();
```

Comment:
This probably should be a thread-safe implementation of some kind, to be a robust implementation of `ITestLogerWithParameters`.

The one and only use right now is from `BoogieXmlConvertor` so not technically necessary, but especially if we move away from the `/xml`-based implementation we could easily want to handle concurrent `TestResultHandler` invocations.